Add user attribute documentation (#2439)
aws-amplify/docs: docs/lib/auth/fragments/android/user_attributes/20_update_user_attribute.md

To update a single user attribute, call `updateUserAttribute`:

<amplify-block-switcher>
<amplify-block name="Java">

```java

> Amplify.Auth.updateUserAttribute(new AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com"),
>     result -> Log.i("Update user attribute succeeded", "Result: " + result.toString()),
>     error -> Log.e("Update user attribute failed", error.toString())
> );

Reviewer: ```suggestion
AuthUserAttribute userEmail =
    new AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com");
Amplify.Auth.updateUserAttribute(userEmail,
    result -> Log.i("AuthDemo", "Updated user attribute = " + result.toString()),
    error -> Log.e("AuthDemo", "Failed to update user attribute.", error)
);
```

```

</amplify-block>
<amplify-block name="Kotlin">

```kotlin

> Amplify.Auth.updateUserAttribute(
>     AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com"),
>     { result-> Log.i("Update user attribute succeeded", "Result: " + result.toString()) },
>     { error-> Log.e("Update user attribute failed", error.toString()) }

Reviewer: ```suggestion
val userEmail = AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com")
Amplify.Auth.updateUserAttribute(userEmail,
    { Log.i("AuthDemo", "Updated user attribute = $it") },
    { Log.e("AuthDemo", "Failed to update user attribute.", $it) }
```

)
```

</amplify-block>
<amplify-block name="RxJava">

```java

> RxAmplify.Auth.updateUserAttribute(AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com"))
>     .subscribe(
>         result -> Log.i("Update user attribute succeeded", "Result: " + result.toString()),
>         error -> Log.e("Update user attribute failed", error.toString())
>     );

Reviewer: ```suggestion
AuthUserAttribute userEmail =
    new AuthUserAttribute(AuthUserAttributeKey.email(), "email@email.com");
RxAmplify.Auth.updateUserAttribute(userEmail)
    .subscribe(
        result -> Log.i("AuthDemo", "Updated user attribute = " + result.toString()),
        error -> Log.e("AuthDemo", "Failed to update user attribute.", error)
    );
```

```

</amplify-block>
</amplify-block-switcher>

To update multiple user attributes at a time, call `updateUserAttributes`:

<amplify-block-switcher>
<amplify-block name="Java">

```java
Amplify.Auth.updateUserAttributes(
    attributes, // attributes is a list of AuthUserAttribute

>     result -> Log.i("Update user attributes succeeded", "Result: " + result.toString()),
>     error -> Log.e("Update user attributes failed", error.toString())

Reviewer: ```suggestion
    result -> Log.i("AuthDemo", "Updated user attributes = " + result.toString()),
    error -> Log.e("AuthDemo", "Failed to update user attributes.", error.)
```

);
```

</amplify-block>
<amplify-block name="Kotlin">

```kotlin
Amplify.Auth.updateUserAttributes(
    attributes, // attributes is a list of AuthUserAttribute

>     { result-> Log.i("Update user attributes succeeded", "Result: " + result.toString()) },
>     { error-> Log.e("Update user attributes failed", error.toString()) }

Reviewer: ```suggestion
    { Log.i("Updated user attributes = $it") },
    { Log.e("Failed to update user attributes.", $it) }
```

)
```

</amplify-block>
<amplify-block name="RxJava">

```java
// attributes is a list of AuthUserAttribute

> RxAmplify.Auth.updateUserAttributes(attributes)
>     .subscribe(
>         result -> Log.i("Update user attributes succeeded", "Result: " + result.toString()),
>         error -> Log.e("Update user attributes failed", error.toString())
>     );

Reviewer: ```suggestion
RxAmplify.Auth.updateUserAttributes(attributes)
    .subscribe(
        result -> Log.i("AuthDemo", "Updated user attributes = " + result.toString()),
        error -> Log.e("AuthDemo", "Failed to update user attributes.", error)
    );
```
